gdctools/gdc_dice.py

```diff
@@ -380,32 +395,49 @@ def dice_one(file_dict, translation_dict, mirror_proj_root, diced_root,
                 already_diced = all(os.path.isfile(p) for p in expected_paths)
                 if force or not already_diced:
                     logging.info("Dicing file " + mirror_path)
-                    convert(file_dict, mirror_path, dice_path)
+                    try:
+                        convert(file_dict, mirror_path, dice_path)
+                        dice_one_status = 'pass'
+                    except Exception as e:
+                        logging.warning('Dice converter failed: %s'%str(e))
                 else:
                     logging.info("Skipping file " + mirror_path + " (already diced)")
+                    dice_one_status = 'cached'
 
-                append_diced_metadata(file_dict, expected_paths,
+                duplicate_detected = append_diced_metadata(file_dict, expected_paths,
```

Comment:
This seems to return the opposite value of the spirit of the call:  by returning True for the bad condition when the append failed (due to having a duplicate); this prevents the code from passing review